api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationSearchAdviserTest.java

```diff
@@ -0,0 +1,450 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class LocationSearchAdviserTest extends AOPContextSensitiveTest {
+
+    private LocationService locationService;
+
+    private static final String LOCATION_TEST_DATA_XML_LOCATION = "include/LocationTestData.xml";
+
+    private Location location1;
+    private Location location2;
+    private Location location3;
+    private Location location4;
+    private Location location5;
+
+    private static final String SEARCH_QUERY_MATCH_ALL_LOCATIONS = "test location";
+    private static final String TAG_NAME = "Test Location Tag";
+
+    private Set<String> restrictedGetMethodNames = new HashSet<String>();
+
+    public LocationSearchAdviserTest() {
+        restrictedGetMethodNames.add("getDefaultLocation");
+        restrictedGetMethodNames.add("getDefaultLocationFromSting");
+        restrictedGetMethodNames.add("getLocationByUuid");
+        restrictedGetMethodNames.add("getAllLocations");
+        restrictedGetMethodNames.add("getLocations");
+        restrictedGetMethodNames.add("getLocationsByTag");
+        restrictedGetMethodNames.add("getRootLocations");
+    }
+
+    @Override
+    protected void setInterceptorAndServices(TestWithAOP testCase) {
+        testCase.setInterceptor(new LocationServiceInterceptorAdvice(restrictedGetMethodNames));
+        testCase.addService(LocationService.class);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        locationService = Context.getLocationService();
+        executeDataSet(LOCATION_TEST_DATA_XML_LOCATION);
+
+        location1 = locationService.getLocation(1);
+        location2 = locationService.getLocation(2);
+        location3 = locationService.getLocation(3);
+        location4 = locationService.getLocation(4);
+        location5 = locationService.getLocation(5);
+        assertNotNull(location1);
+        assertNotNull(location2);
+        assertNotNull(location3);
+        assertNotNull(location4);
+        assertNotNull(location5);
+    }
+
+    @Test
+    public void getDefaultLocation_adminShouldGetDefaultLocation() {
+        setDefaultLocationGP(location2);
+        authenticateAdmin();
+        assertAuthenticatedAdmin();
+
+        Location result = locationService.getDefaultLocation();
+        assertEquals(location2, result);
+    }
+
+    @Test
+    public void getDefaultLocation_nonAdminWithLocationPropertyShouldGetOnlyDefaultLocationIfItsAccessLocation() {
+        setDefaultLocationGP(location3);
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        Location result = locationService.getDefaultLocation();
+        assertEquals(location3, result);
+    }
+
+    @Test
+    public void getDefaultLocation_nonAdminWithLocationPropertyShouldNotGetDefaultLocationIfItsNotAccessLocation() {
```

Comment:
Change the method name as mentioned above. It may easy to understand the test use case

Have you checked the user access location for null here?